scripts/Connector/Connector_JST/conn_jst_XA_vertical.py

```diff
@@ -0,0 +1,251 @@
+            ]
+            kicad_mod.append(PolygoneLine(polygone=poly_silk_inner_outline, layer='F.SilkS', width=configuration['silk_line_width']))
+
+        #cut in short sides
+        kicad_mod.append(Line(start=[silk_x_min, y_max - 3.2], end=[silk_inner_left, y_max - 3.2], layer='F.SilkS', width=configuration['silk_line_width']))
+        kicad_mod.append(Line(start=[silk_x_max, y_max - 3.2], end=[silk_inner_right, y_max - 3.2], layer='F.SilkS', width=configuration['silk_line_width']))
+
+
+    ########################### Pin 1 marker ################################
+    poly_pin1_marker = [
+        {'x':silk_x_min-pin1_marker_offset+pin1_marker_linelen, 'y':silk_y_min-pin1_marker_offset},
+        {'x':silk_x_min-pin1_marker_offset, 'y':silk_y_min-pin1_marker_offset},
+        {'x':silk_x_min-pin1_marker_offset, 'y':silk_y_min-pin1_marker_offset+pin1_marker_linelen}
+    ]
+    kicad_mod.append(PolygoneLine(polygone=poly_pin1_marker, layer='F.SilkS', width=configuration['silk_line_width']))
+    if fab_pin1_marker_type == 1:
+        kicad_mod.append(PolygoneLine(polygone=poly_pin1_marker, layer='F.Fab', width=configuration['fab_line_width']))
+
+    if fab_pin1_marker_type == 2:
+        poly_pin1_marker_type2 = [
+            {'x':-1, 'y':y_min},
+            {'x':0, 'y':y_min+1},
+            {'x':1, 'y':y_min}
+        ]
+        kicad_mod.append(PolygoneLine(polygone=poly_pin1_marker_type2, layer='F.Fab', width=configuration['fab_line_width']))
+
+    ########################## Fab Outline ###############################
+    kicad_mod.append(RectLine(start=[x_min,y_min], end=[x_max,y_max],
+        layer='F.Fab', width=configuration['fab_line_width']))
+    ############################# CrtYd ##################################
+    part_x_min = x_min
+    part_x_max = x_max
+    part_y_min = y_min
+    part_y_max = y_max
+
+    cx1 = roundToBase(part_x_min-configuration['courtyard_offset']['connector'], configuration['courtyard_grid'])
+    cy1 = roundToBase(part_y_min-configuration['courtyard_offset']['connector'], configuration['courtyard_grid'])
+
+    cx2 = roundToBase(part_x_max+configuration['courtyard_offset']['connector'], configuration['courtyard_grid'])
+    cy2 = roundToBase(part_y_max+configuration['courtyard_offset']['connector'], configuration['courtyard_grid'])
+
+    kicad_mod.append(RectLine(
+        start=[cx1, cy1], end=[cx2, cy2],
+        layer='F.CrtYd', width=configuration['courtyard_line_width']))
+
+
+    ############################# Pads ##################################
+    pad_size = [pitch - pad_to_pad_clearance, drill_size + 2*pad_copper_y_solder_length]
+    if pad_size[0] - drill_size < 2*min_annular_ring:
+        pad_size[0] = drill_size + 2*min_annular_ring
+
+    # kicad_mod.append(Pad(number=1, type=Pad.TYPE_THT, shape=Pad.SHAPE_RECT,
+    #                     at=[0, 0], size=pad_size,
+    #                     drill=drill_size, layers=Pad.LAYERS_THT))
```

Comment:
this comment is probably not needed anymore either

Reply:
ok. removed